src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php

```diff
@@ -87,16 +90,24 @@ public function suggest($pattern)
     }
 
     /**
-     * @{inheritDoc}
+     * {@inheritDoc}
      */
     public function all(AuthorizationCheckerInterface $securityContext)
     {
         $ret = array();
-        foreach ($this->providers as $provider) {
+        foreach ($this->getProviders() as $provider) {
             if ($provider instanceof ListableProvider) {
                 $ret = array_merge($ret, $provider->all($securityContext));
             }
         }
         return $ret;
     }
+
+    /**
+     * @return Provider[]|array
+     */
+    private function getProviders()
+    {
+        return iterable($this->providers)->sorted('priority')->map('provider')->values();
+    }
```

Comment:
When we sort during the `addProvider`, then we will not need this anymore, we can just iterate over `$this->providers` again